File: qc-domain/src/main/java/com/eager/questioncloud/domain/coupon/model/Coupon.java
package com.eager.questioncloud.domain.coupon.model;

import com.eager.questioncloud.domain.coupon.vo.CouponType;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Coupon {
    private Long id;
    private String code;
    private String title;
    private CouponType couponType;
    private int value;
    private int remainingCount;
    private LocalDateTime endAt;

    @Builder
    public Coupon(Long id, String code, String title, CouponType couponType, int value, int remainingCount, LocalDateTime endAt) {
        this.id = id;
        this.code = code;
        this.title = title;
        this.couponType = couponType;
        this.value = value;
        this.remainingCount = remainingCount;
        this.endAt = endAt;
    }

    public void decrease() {
        this.remainingCount = this.remainingCount - 1;
    }
}

File: qc-payment/src/main/kotlin/com/eager/questioncloud/coupon/domain/Coupon.kt
package com.eager.questioncloud.coupon.domain

import com.eager.questioncloud.coupon.enums.CouponType
import java.time.LocalDateTime

class Coupon(
    var id: Long = 0,
    var code: String,
    var title: String,
    var couponType: CouponType,
    var value: Int,
    var remainingCount: Int,
    var endAt: LocalDateTime
)
